File: java/poker/src/main/java/com/duchen/poker/model/Player.java
package com.duchen.poker.model;

import java.util.List;

public class Player {

    public enum Role {
        BOSS, FARMER_NEXT, FARMER_PRE
    }

    Role mRole;
    HandCardData mHandCardData;

    public Player() {
        mHandCardData = new HandCardData();
    }

    public void setRole(Role role) {
        mRole = role;
    }

    public Role getRole() {
        return mRole;
    }

    public boolean askToBeBoss() {
        return true;
    }

    public void setCards(List<Integer> cards) {
        mHandCardData.setHandCardList(cards);
    }

    public void becomeBoss(List<Integer> threeCards) {
        mHandCardData.addCards(threeCards);
    }

    @Override
    public String toString() {
        return "Player{" + "mRole=" + mRole + ", mHandCardData=" + mHandCardData + '}';
    }
}

File: java/poker/src/main/kotlin/poker/model/Player.kt
package com.duchen.poker.model

class Player {

    var role : Role
    internal var mHandCardData: HandCardData

    enum class Role {
        BOSS, FARMER_NEXT, FARMER_PRE
    }

    init {
        role = Role.BOSS
        mHandCardData = HandCardData()
    }

    fun askToBeBoss(): Boolean {
        return true
    }

    fun setCards(cards: MutableList<Int>) {
        mHandCardData.handCardList = cards
    }

    fun becomeBoss(threeCards: List<Int>) {
        mHandCardData.addCards(threeCards)
    }

    override fun toString(): String {
        return "Player{mRole=$role, mHandCardData=$mHandCardData}"
    }
}
